src/App.js

```diff
@@ -1,15 +1,204 @@
+    let vtBorderData = L.geoJSON(borderData);
+    //loop ensures marker will be within state border
+    while (layerLength !== 1) {
+      //Generates random latitude and longitude points
+      latRandGen = Math.random() * (vtMaxLat - vtMinLat) + vtMinLat;
+      longRandGen = Math.random() * (vtMaxLong - vtMinLong) + vtMinLong;
+      //Vermont is a polygon shaped state
+      //leaflet-pip is a library for finding out whether a point is inside a polygon
+      // checking if the randomly generated co-ords are within the vermont border points
+      layerLength = leafletPip.pointInLayer(
+        [longRandGen, latRandGen],
+        vtBorderData
+      ).length;
+    }
+    //assigns random points to our latitude and longitude variables
+    setLatRandom(latRandGen);
+    setLongRandom(longRandGen);
+    setCenter([latRandGen, longRandGen]);
+  }
+  //places the map marker in a random spot as well as disables start button and enables guess and quit buttons
+  function startClickHandler() {
+    setStart(false);
+    setButtonState(!buttonState);
+    setZoom(18);
+    RandomStart();
+  }
+
+  //triggers the guess modal box to appear
+  function guessClickHandler() {
+    setGuessBox(!guessBox);
+  }
+
+  //Displays the players location and informs them of the county and town the marker has been placed in
+  function quitClickHandler() {
+    setButtonState(!buttonState)
+
+    setQuit(true);
+  }
+
+  //function that returns player to the initial random spot
+  function returnToStart() {
+    setCenter([
+      latRandom + moveSouthCount * 0.002 - moveNorthCount * 0.002,
+      longRandom + moveWestCount * 0.002 - moveEastCount * 0.002,
+    ]);
+    setMoveNorthCount(0);
+    setMoveSouthCount(0);
+    setMoveWestCount(0);
+    setMoveEastCount(0);
+  }
 
-  const [center, setCenter] = useState([43.88, -72.7317])
+  //function to move marker north when north button is pressed, for some reason directional buttons must be pressed twice before the changes are made
+  function moveNorth() {
+    setMoveNorthCount(moveNorthCount + 1);
+    setLatRandom(latRandom + 0.002);
+    setCenter([latRandom, longRandom]);
+    setScore(score - 1);
+  }
 
+  //function to marker south when south button is pressed
+  function moveSouth() {
+    setMoveSouthCount(moveSouthCount + 1);
+    setLatRandom(latRandom - 0.002);
+    setCenter([latRandom, longRandom]);
+    setScore(score - 1);
+  }
+
+  //function to move marker east when east button is pressed
+  function moveEast() {
+    setMoveEastCount(moveEastCount + 1);
+    setLongRandom(longRandom + 0.002);
+    setCenter([latRandom, longRandom]);
+    setScore(score - 1);
+  }
+
+  //function to move marker west when west button is pressed
+  function moveWest() {
+    setMoveWestCount(moveWestCount + 1);
+    setLongRandom(longRandom - 0.002);
+    setCenter([latRandom, longRandom]);
+    setScore(score - 1);
+  }
```

Comment:
It appears that the issue with the initial button click is due to an issue with props and state. It looks like the buttons are updating state correctly, but because of the props being passed the state is not initially updating correctly.